### ISSUE TITLE
Failed to compile ktor client 

### ISSUE THREAD

**Issue by e5l** [JetBrains/kotlin-native#1761]
Reproduce: 
1. use branch https://github.com/ktorio/ktor/tree/ios-client-engine
2. `./gradlew clean compileKonan`
```
error: compilation failed: Non-local callable reference to suspend lambda: private final suspend fun kotlinx.coroutines.experimental.io.WriterScope.`execute$<anonymous>_8`(`$this`: io.ktor.client.engine.ios.IosClientEngine.execute.<anonymous>_4.<no name provided>): kotlin.Unit defined in io.ktor.client.engine.ios.IosClientEngine[SimpleFunctionDescriptorImpl@61defefb]

 * Source files: Ios.kt, IosClientEngine.kt, IosHttpResponse.kt, Utils.kt
 * Compiler version info: Konan: 0.8-dev-2538 / Kotlin: 1.2.60
 * Output kind: LIBRARY

exception: java.lang.Error: Non-local callable reference to suspend lambda: private final suspend fun kotlinx.coroutines.experimental.io.WriterScope.`execute$<anonymous>_8`(`$this`: io.ktor.client.engine.ios.IosClientEngine.execute.<anonymous>_4.<no name provided>): kotlin.Unit defined in io.ktor.client.engine.ios.IosClientEngine[SimpleFunctionDescriptorImpl@61defefb]
        at org.jetbrains.kotlin.backend.konan.lower.SuspendFunctionsLowering$transformCallableReferencesToSuspendLambdas$$inlined$forEach$lambda$1.visitFunctionReference(SuspendFunctionsLowering.kt:108)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitFunctionReference(IrElementTransformerVoid.kt:180)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitFunctionReference(IrElementTransformerVoid.kt:24)
        at org.jetbrains.kotlin.ir.expressions.impl.IrFunctionReferenceImpl.accept(IrFunctionReferenceImpl.kt:86)
        at org.jetbrains.kotlin.ir.expressions.IrExpression$DefaultImpls.transform(IrExpression.kt:27)
        at org.jetbrains.kotlin.ir.expressions.impl.IrExpressionBase.transform(IrExpressionBase.kt:23)
        at org.jetbrains.kotlin.ir.expressions.impl.IrExpressionBase.transform(IrExpressionBase.kt:23)
        at org.jetbrains.kotlin.ir.expressions.impl.IrContainerExpressionBase.transformChildren(IrContainerExpressionBase.kt:42)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:260)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildrenVoid(IrElementTransformerVoid.kt:255)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildren(IrElementTransformerVoid.kt:25)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:102)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitContainerExpression(IrElementTransformerVoid.kt:114)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:117)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:118)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlock(IrElementTransformerVoid.kt:24)
        at org.jetbrains.kotlin.ir.expressions.impl.IrBlockImpl.accept(IrBlockImpl.kt:38)
        at org.jetbrains.kotlin.ir.expressions.IrExpression$DefaultImpls.transform(IrExpression.kt:27)
        at org.jetbrains.kotlin.ir.expressions.impl.IrExpressionBase.transform(IrExpressionBase.kt:23)
        at org.jetbrains.kotlin.ir.expressions.impl.IrCallWithIndexedArgumentsBase.transformChildren(IrCallWithIndexedArgumentsBase.kt:63)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:260)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildrenVoid(IrElementTransformerVoid.kt:255)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildren(IrElementTransformerVoid.kt:25)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitExpression(IrElementTransformerVoid.kt:102)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitMemberAccess(IrElementTransformerVoid.kt:156)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitFunctionAccess(IrElementTransformerVoid.kt:159)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:162)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:163)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitCall(IrElementTransformerVoid.kt:24)
        at org.jetbrains.kotlin.ir.expressions.impl.IrCallImpl.accept(IrCallImpl.kt:151)
        at org.jetbrains.kotlin.ir.expressions.IrExpression$DefaultImpls.transform(IrExpression.kt:27)
        at org.jetbrains.kotlin.ir.expressions.impl.IrExpressionBase.transform(IrExpressionBase.kt:23)
        at org.jetbrains.kotlin.ir.declarations.impl.IrVariableImpl.transformChildren(IrVariableImpl.kt:89)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:260)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildrenVoid(IrElementTransformerVoid.kt:255)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildren(IrElementTransformerVoid.kt:25)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitDeclaration(IrElementTransformerVoid.kt:46)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitVariable(IrElementTransformerVoid.kt:87)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitVariable(IrElementTransformerVoid.kt:88)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitVariable(IrElementTransformerVoid.kt:24)
        at org.jetbrains.kotlin.ir.declarations.impl.IrVariableImpl.accept(IrVariableImpl.kt:82)
        at org.jetbrains.kotlin.ir.declarations.IrDeclaration$DefaultImpls.transform(IrDeclaration.kt:36)
        at org.jetbrains.kotlin.ir.declarations.impl.IrDeclarationBase.transform(IrDeclarationBase.kt:25)
        at org.jetbrains.kotlin.ir.expressions.impl.IrBlockBodyImpl.transformChildren(IrBlockBodyImpl.kt:43)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:260)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildrenVoid(IrElementTransformerVoid.kt:255)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.transformChildren(IrElementTransformerVoid.kt:25)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBody(IrElementTransformerVoid.kt:90)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:96)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:97)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoid.visitBlockBody(IrElementTransformerVoid.kt:24)
        at org.jetbrains.kotlin.ir.expressions.impl.IrBlockBodyImpl.accept(IrBlockBodyImpl.kt:34)
        at org.jetbrains.kotlin.ir.expressions.IrBody$DefaultImpls.transform(IrBody.kt:24)
        at org.jetbrains.kotlin.ir.expressions.IrBlockBody$DefaultImpls.transform(IrBody.kt)
        at org.jetbrains.kotlin.ir.expressions.impl.IrBlockBodyImpl.transform(IrBlockBodyImpl.kt:26)
        at org.jetbrains.kotlin.ir.declarations.impl.IrFunctionBase.transformChildren(IrFunctionBase.kt:67)
        at org.jetbrains.kotlin.ir.visitors.IrElementTransformerVoidKt.transformChildrenVoid(IrElementTransformerVoid.kt:260)
        at org.jetbrains.kotlin.backend.konan.lower.SuspendFunctionsLowering.transformCallableReferencesToSuspendLambdas(SuspendFunctionsLowering.kt:99)
        at org.jetbrains.kotlin.backend.konan.lower.SuspendFunctionsLowering.lower(SuspendFunctionsLowering.kt:76)
        at org.jetbrains.kotlin.backend.common.LowerKt$asClassLoweringPass$1.lower(Lower.kt:64)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:57)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:44)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:50)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:50)
        at org.jetbrains.kotlin.ir.declarations.impl.IrClassImpl.accept(IrClassImpl.kt:85)
        at org.jetbrains.kotlin.ir.declarations.impl.IrClassImpl.acceptChildren(IrClassImpl.kt:90)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoidKt.acceptChildrenVoid(IrElementVisitorVoid.kt:245)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:56)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:44)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:50)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitClass(Lower.kt:50)
        at org.jetbrains.kotlin.ir.declarations.impl.IrClassImpl.accept(IrClassImpl.kt:85)
        at org.jetbrains.kotlin.ir.declarations.impl.IrFileImpl.acceptChildren(IrFileImpl.kt:71)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoidKt.acceptChildrenVoid(IrElementVisitorVoid.kt:245)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitElement(Lower.kt:52)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitPackageFragment(IrElementVisitorVoid.kt:30)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitPackageFragment(Lower.kt:50)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:37)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitFile(Lower.kt:50)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitFile(IrElementVisitorVoid.kt:38)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitFile(Lower.kt:50)
        at org.jetbrains.kotlin.backend.common.LowerKt$runOnFilePostfix$1.visitFile(Lower.kt:50)
        at org.jetbrains.kotlin.ir.declarations.impl.IrFileImpl.accept(IrFileImpl.kt:68)
        at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoidKt.acceptVoid(IrElementVisitorVoid.kt:241)
        at org.jetbrains.kotlin.backend.common.LowerKt.runOnFilePostfix(Lower.kt:50)
        at org.jetbrains.kotlin.backend.common.LowerKt.runOnFilePostfix(Lower.kt:69)
        at org.jetbrains.kotlin.backend.konan.KonanLower$lowerFile$18.invoke(KonanLower.kt:155)
        at org.jetbrains.kotlin.backend.konan.KonanLower$lowerFile$18.invoke(KonanLower.kt:35)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:139)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:118)
        at org.jetbrains.kotlin.konan.util.UtilKt.profileIf(Util.kt:34)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:138)
        at org.jetbrains.kotlin.backend.konan.KonanLower.lowerFile(KonanLower.kt:154)
        at org.jetbrains.kotlin.backend.konan.KonanLower.lower(KonanLower.kt:45)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4$1.invoke(KonanDriver.kt:92)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4$1.invoke(KonanDriver.kt)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:139)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:118)
        at org.jetbrains.kotlin.konan.util.UtilKt.profileIf(Util.kt:34)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:138)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4.invoke(KonanDriver.kt:91)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt$runTopLevelPhases$4.invoke(KonanDriver.kt)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:139)
        at org.jetbrains.kotlin.backend.konan.PhaseManager$phase$$inlined$with$lambda$1.invoke(KonanPhases.kt:118)
        at org.jetbrains.kotlin.konan.util.UtilKt.profileIf(Util.kt:34)
        at org.jetbrains.kotlin.backend.konan.PhaseManager.phase$backend_native_compiler(KonanPhases.kt:138)
        at org.jetbrains.kotlin.backend.konan.KonanDriverKt.runTopLevelPhases(KonanDriver.kt:90)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:75)
        at org.jetbrains.kotlin.cli.bc.K2Native.doExecute(K2Native.kt:42)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:95)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.java:50)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:88)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:66)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:34)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMainNoExit(CLITool.kt:180)
        at org.jetbrains.kotlin.cli.common.CLITool$Companion.doMain(CLITool.kt:171)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:197)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion$main$1.invoke(K2Native.kt:188)
        at org.jetbrains.kotlin.konan.util.UtilKt.profileIf(Util.kt:34)
        at org.jetbrains.kotlin.konan.util.UtilKt.profile(Util.kt:29)
        at org.jetbrains.kotlin.cli.bc.K2Native$Companion.main(K2Native.kt:190)
        at org.jetbrains.kotlin.cli.bc.K2NativeKt.main(K2Native.kt:202)
        at org.jetbrains.kotlin.cli.utilities.MainKt.main(main.kt:27)


> Task :ktor-client:ktor-client-ios:compileKonanKtor_client_iosIos_arm64 FAILED
:ktor-client:ktor-client-ios:compileKonanKtor_client_iosIos_arm64 (Thread[Task worker for ':' Thread 2,5,main]) completed. Took 8.801 secs.

FAILURE: Build failed with an exception.
```

**Comment by soywiz** [comment]
https://github.com/JetBrains/kotlin-native/issues/1724 Is this one? Have you tried 0.8-dev-2632 (probably fixed there already) / master?

I was able to try master with gradle by downloading kotlin-native repo at the same level of my project, and then run this:
```
export KONAN_REPO=$PWD/../kotlin-native

pushd $KONAN_REPO && git pull && ./gradlew clean dependencies:update dist distPlatformLibs && popd # run this once since it is costly

./gradlew check -Pkonan.home=$KONAN_REPO/dist --include-build $KONAN_REPO/shared --include-build $KONAN_REPO/tools/kotlin-native-gradle-plugin
```

**Comment by e5l** [comment]
Fixed in  0.8-dev-2701
